File: android/src/main/java/top/huic/tencent_im_plugin/listener/CustomConversationListener.java
package top.huic.tencent_im_plugin.listener;

import com.tencent.imsdk.v2.V2TIMConversation;
import com.tencent.imsdk.v2.V2TIMConversationListener;

import java.util.ArrayList;
import java.util.List;

import top.huic.tencent_im_plugin.TencentImPlugin;
import top.huic.tencent_im_plugin.entity.CustomConversationEntity;
import top.huic.tencent_im_plugin.enums.ListenerTypeEnum;

/**
 * 自定义会话监听
 */
public class CustomConversationListener extends V2TIMConversationListener {
    /**
     * 同步服务开始
     */
    @Override
    public void onSyncServerStart() {
        super.onSyncServerStart();
        TencentImPlugin.invokeListener(ListenerTypeEnum.SyncServerStart, null);
    }

    /**
     * 同步服务完成
     */
    @Override
    public void onSyncServerFinish() {
        super.onSyncServerFinish();
        TencentImPlugin.invokeListener(ListenerTypeEnum.SyncServerFinish, null);
    }

    /**
     * 同步服务失败
     */
    @Override
    public void onSyncServerFailed() {
        super.onSyncServerFailed();
        TencentImPlugin.invokeListener(ListenerTypeEnum.SyncServerFailed, null);
    }

    /**
     * 新会话
     */
    @Override
    public void onNewConversation(List<V2TIMConversation> conversationList) {
        super.onNewConversation(conversationList);
        List<CustomConversationEntity> data = new ArrayList<>(conversationList.size());
        for (V2TIMConversation v2TIMConversation : conversationList) {
            data.add(new CustomConversationEntity(v2TIMConversation));
        }
        TencentImPlugin.invokeListener(ListenerTypeEnum.NewConversation, data);
    }

    /**
     * 会话刷新
     */
    @Override
    public void onConversationChanged(List<V2TIMConversation> conversationList) {
        super.onConversationChanged(conversationList);
        List<CustomConversationEntity> data = new ArrayList<>(conversationList.size());
        for (V2TIMConversation v2TIMConversation : conversationList) {
            data.add(new CustomConversationEntity(v2TIMConversation));
        }
        TencentImPlugin.invokeListener(ListenerTypeEnum.ConversationChanged, data);
    }
}

File: ios/Classes/listener/CustomConversationListener.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK

/// 自定义会话监听
class CustomConversationListener: NSObject, V2TIMConversationListener {
    /// 同步服务开始
    func onSyncServerStart() {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.SyncServerStart, params: nil)
    }

    /// 同步服务完成
    func onSyncServerFinish() {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.SyncServerFinish, params: nil)
    }

    /// 同步服务失败
    func onSyncServerFailed() {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.SyncServerFailed, params: nil)
    }

    /// 新会话
    func onNewConversation(_ conversationList: [V2TIMConversation]!) {
        var cs: [[String: Any]] = [];
        for item in conversationList {
            cs.append(CustomConversationEntity.getDict(info: item));
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.NewConversation, params: cs)
    }

    /// 会话刷新
    func onConversationChanged(_ conversationList: [V2TIMConversation]!) {
        var cs: [[String: Any]] = [];
        for item in conversationList {
            cs.append(CustomConversationEntity.getDict(info: item));
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.ConversationChanged, params: cs)
    }
}
